### PR TITLE
Fix CA2247 handling of base(TaskContinuationOptions)

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/roslyn-analyzers/issues/5135

### CODE REVIEW

File: src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Tasks/DoNotCreateTaskCompletionSourceWithWrongArguments.cs [hunk 1]
```diff
@@ -41,19 +41,37 @@ public override void Initialize(AnalysisContext context)
...
+
+                        if (conversionOperation is not null)
                         {
                             operationContext.ReportDiagnostic(conversionOperation.CreateDiagnostic(Rule));
                         }
-                    }, OperationKind.ObjectCreation);
+                    }, OperationKind.ObjectCreation, OperationKind.Invocation);
+
+                    IConversionOperation? MatchInvalidContinuationOptions(IMethodSymbol targetMethod, ImmutableArray<IArgumentOperation> arguments) =>
+                        (targetMethod.ContainingType.OriginalDefinition.Equals(tcsGenericType) || (tcsType != null && targetMethod.ContainingType.OriginalDefinition.Equals(tcsType))) &&
```
Comment (Youssef1313) [inline src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Tasks/DoNotCreateTaskCompletionSourceWithWrongArguments.cs:67]: nit: I see this is existing before the change, but null check seems redundant here. `Equals` call should already return false.

**Review by buyaa-n — APPROVED** [review]
LGTM

**Comment by stephentoub** [comment]
@jmarolf , is this the right branch, or should I target it to a different one?

**Comment by jmarolf** [comment]
@stephentoub main will merge into the dotnet 6 branch automatically. unless there is a specific preview you want this to go in main is the right branch.

**Review by jmarolf — APPROVED** [review]

**Comment by stephentoub** [comment]
> main will merge into the dotnet 6 branch automatically

Cool, thanks.

**Pull request merged**